[weiwong834] iP

weiwong834 opened the pull request:
### **Squid 🐙 - The Task Wrangler** 
> “Another day, another migraine” – Squidward ([source](https://mustreadquotes.com/quote/ctlen0p0fvntek/))

Squid is the _ultimate_ task wrangler that's obsessed with keeping your life organised. It can:

- create and manage your to-do list
- keep track of deadlines
- organise events
- search through tasks easily

Features:

- [x] Managing tasks
- [ ] Reminders

All you need to do is,

1. download it from here
2. double-click it
3. add your tasks
4. let it manage your tasks for you 🌊 

If you are a Java programmer, you can use it to practice Java too. Here's the ```main``` method:
```Java
public class Squid{
    public static void main(String[] args) {
        ...
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

weiwong834 committed:
Renamed Duke to Squid, added greeting + goodbye

weiwong834 committed:
echo

weiwong834 committed:
Add, list

weiwong834 committed:
added Task class

weiwong834 committed:
Mark as done

weiwong834 committed:
Added subclasses for Task

weiwong834 committed:
ToDos, Events, Deadlines

weiwong834 committed:
edited I/O testing files + squid

weiwong834 committed:
added error handling

weiwong834 committed:
added delete (task #)

weiwong834 committed:
added enums in Squid.java

weiwong834 committed:
Add ability to save and load list of tasks

weiwong834 committed:
Add task directory

weiwong834 committed:
Add comment in Squid.java

weiwong834 committed:
Merge branch 'branch-Level-7'

weiwong834 committed:
Update classes to use LocalDate

weiwong834 committed:
Add functionality to set time for deadline/ event

weiwong834 committed:
Add show command

weiwong834 committed:
Merge branch 'branch-Level-8'

weiwong834 committed:
Rename TaskStorage.java to Storage.java

weiwong834 committed:
Add OOP implementations

weiwong834 committed:
Add packages

weiwong834 committed:
Update squid.txt

weiwong834 committed:
Merge branch 'add-gradle-support'

gekjunxu commented on src/main/java/squid/Parser.java:
```diff
@@ -0,0 +1,104 @@
+package squid;
+
+import exceptions.SquidException;
+import command.*;
```
Maybe you can consider not using wildcard imports?

gekjunxu commented on src/main/java/squid/Squid.java:
```diff
@@ -0,0 +1,37 @@
+import command.Command;
+import command.ExitCommand;
+import exceptions.SquidException;
+import task.*;
```
Maybe consider not using wildcard imports?

gekjunxu commented on src/main/java/squid/Ui.java:
```diff
@@ -0,0 +1,107 @@
+
+    public void showTasksOnDate(LocalDate date, TaskList tasks) {
+        System.out.println("Task on " + date + ":");
+        boolean found = false;
```
Perhaps can use the isFound naming, and to specifiy what is found?

gekjunxu commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String toFileFormat() {
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + by.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```
Maybe consider separating into 2 lines? seems to be over 120 chars here

gekjunxu commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:mm")) + ")";
```
Maybe consider separating into 2 lines? might be over 120 chars here

gekjunxu commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+        this.isDone = isDone;
+    }
+
+    public LocalDateTime getFrom() {
```
Maybe can use something like getFromDate or something to better show its getting date/time?

gekjunxu commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,61 @@
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+        switch (type) {
+            case "T":
```
Might want to update the indentation for the case statements

gekjunxu reviewed (commented):
Looks good mostly, other than some nits 👍

weiwong834 committed:
Add isDone() in Task

weiwong834 committed:
Add JUnit Tests

weiwong834 committed:
Add methods and JavaDoc in Parser

weiwong834 committed:
Add JavaDoc in command and squid packages

matthanfoo commented on src/main/java/squid/Ui.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    public void showLine() {
+        System.out.println("____________________________________________________________");
```
indenting issues here

matthanfoo commented on src/main/java/squid/Ui.java:
```diff
@@ -0,0 +1,107 @@
+        System.out.println("Task on " + date + ":");
+        boolean found = false;
+        for (Task task : tasks.getAllTasks()) {
+            if (task instanceof Deadline && ((Deadline) task).getBy().toLocalDate().equals(date)) {
```
indenting issues here as well affecting readability

matthanfoo commented on src/main/java/squid/Ui.java:
```diff
@@ -0,0 +1,107 @@
+            }
+            if (task instanceof Event) {
+                Event event = (Event) task;
+                if (!event.getFrom().toLocalDate().isAfter(date) && !event.getTo().toLocalDate().isBefore(date)) {
```
similar indenting issues

matthanfoo reviewed (commented):
most of the issues were spotted by the other reviewer, other than that really nice code!

weiwong834 committed:
Update to follow coding standard

weiwong834 committed:
Update build.gradle configuration

weiwong834 committed:
Add Find command

weiwong834 committed:
Merge branch 'branch-A-JavaDoc'

weiwong834 committed:
Merge branch 'branch-A-CodingStandard'

weiwong834 committed:
Merge branch 'branch-Level-9'

weiwong834 committed:
Add Checkstyle

weiwong834 committed:
Update Parser to include FindCommand

weiwong834 committed:
Add GUI configuration

weiwong834 committed:
Add exception class and IncorrectCommand

weiwong834 committed:
Update Parser and Command execute methods to return string

weiwong834 committed:
Merge branch 'branch-Level-10'

weiwong834 committed:
Implement assertions and update build configuration

Assert added to Parser to ensure input strings are not null,
helping to catch any misuse of Parser.parse method.

Enabled assertions in build.gradle to catch logical errors early.

These updates needed for maintaining error handling.

weiwong834 committed:
Refactor isTaskOnDate method for clarity in ShowCommand

Current implementation combines multiple logical conditions in a
single return statement, which can be too complicated.

Changes include:
- Splitting logical conditions into Boolean variables.
- Each variable clearly represents what condition is being checked.
- Using these variables to simplify the final return statement.

This approach avoids complicated expressions, improving readability
and it is easier to debug or modify method in the future.

weiwong834 committed:
Merge pull request #2 from weiwong834/branch-A-Assertions

Implement assertions and update build configuration

weiwong834 committed:
Merge branch 'master' into branch-A-CodeQuality

weiwong834 committed:
Merge pull request #3 from weiwong834/branch-A-CodeQuality

Refactor isTaskOnDate method for clarity in ShowCommand

weiwong834 committed:
Enhance command parsing with alias

Current command parsing requires users to type full command words,
which can be cumbersome and error-prone.

Changes made:
- Added alias support by incorporating a Map.
- Updated `parse` method to handle new command aliases and feedback
for unknown commands.

Short aliases like 't' for `todo` make the application
quicker to use.

This update simplifies user interactions.

weiwong834 committed:
Merge branch 'branch-BCD-Extension'

weiwong834 committed:
Add title and icon to GUI

weiwong834 committed:
Fix checkstyle

weiwong834 committed:
Implement GUI termination feature

weiwong834 committed:
Update GUI design

weiwong834 committed:
Update Chatbot reply

weiwong834 committed:
Add User Guide for all commands

weiwong834 committed:
Add image and table to README.md

weiwong834 committed:
Update Ui.png to crop borders
